plugins/woocommerce/src/Internal/AddressProvider/AbstractAutomatticAddressProvider.php

```diff
@@ -0,0 +1,191 @@
+	}
+
+	/**
+	 * Sets the JWT for the address service.
+	 *
+	 * @param string $jwt The JWT for the address service.
+	 */
+	public function set_jwt( $jwt ) {
+		$this->jwt = $jwt;
+		if ( null !== $jwt ) {
+			set_transient( $this->id . 'address_autocomplete_jwt', $jwt, $this->get_jwt_cache_duration( $jwt ) );
+		} else {
+			delete_transient( $this->id . 'address_autocomplete_jwt' );
+		}
+	}
+
+	/**
+	 * Gets the cache duration for the JWT.
+	 *
+	 * @param string $jwt The JWT for the address service.
+	 * @return int The cache duration for the JWT.
+	 */
+	public function get_jwt_cache_duration( $jwt ) {
+		if ( JsonWebToken::shallow_validate( $jwt ) ) {
+			$parts = JsonWebToken::get_parts( $jwt );
+			if ( property_exists( $parts->payload, 'exp' ) ) {
+				return $parts->payload->exp - time();
+			}
+		}
+
+		return DAY_IN_SECONDS;
+	}
+
+	/**
+	 * Deletes the cached token if we disable the autocomplete service or fetches a new one if it's enabled.
+	 *
+	 * @param string $setting If the service is enabled or disabled.
+	 * @return string the setting value.
+	 */
+	public function refresh_cache( $setting ) {
+		if ( wc_string_to_bool( $setting ) ) {
+			$this->load_jwt();
+		} else {
+			$this->set_jwt( null );
+		}
+
+		return $setting;
+	}
+
+	/**
+	 * Return asset URL, copied from WC_Frontend_Scripts::get_asset_url.
+	 *
+	 * @param string $path Assets path.
+	 * @return string
+	 */
+	public static function get_asset_url( $path ) {
+		/**
+		 * Filters the asset URL.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param string $url The asset URL.
+		 * @param string $path The asset path.
+		 * @return string The filtered asset URL.
+		 */
+		return apply_filters( 'woocommerce_get_asset_url', plugins_url( $path, Constants::get_constant( 'WC_PLUGIN_FILE' ) ), $path );
+	}
+
+
+	/**
+	 * Enqueues the checkout script, checks if it's already registered or not so we don't dubplicate, and prints out the JWT to the page to be consumed.
+	 */
+	public function load_scripts() {
+		$suffix  = Constants::is_true( 'SCRIPT_DEBUG' ) ? '' : '.min';
+		$version = Constants::get_constant( 'WC_VERSION' );
+
+		if ( ! wp_script_is( 'a8c-address-autocomplete-service', 'registred' ) ) {
+			wp_register_script( 'a8c-address-autocomplete-service', self::get_asset_url( 'assets/js/frontend/a8c-address-autocomplete-service' . $suffix . '.js' ), array( 'wc-address-autocomplete' ), $version, array( 'strategy' => 'defer' ) );
+		}
+
+		if ( ! wp_script_is( 'a8c-address-autocomplete-service', 'enqueued' ) ) {
+			wp_enqueue_script( 'a8c-address-autocomplete-service' );
+		}
+
+		wp_add_inline_script(
+			'a8c-address-autocomplete-service',
+			sprintf(
+				'var a8cAddressAutocompleteServiceKeys = a8cAddressAutocompleteServiceKeys || {}; a8cAddressAutocompleteServiceKeys[ "%1$s" ] = { key: %2$s, canTelemetry: %3$s };',
+				$this->id,
+				wp_json_encode( $this->get_jwt() ),
+				wp_json_encode( false !== $this->can_telemetry() && (bool) $this->can_telemetry() )
```

Comment:
```suggestion
				wp_json_encode( $this->can_telemetry() )
```
Can we do this? If not what am I missing?

Reply:
We don't know what `can_telemetry` return, it can be an error or something, so I'm trying to be defensive.

Reply:
Makes sense 👍